lmfit/confidence.py

```diff
@@ -183,11 +183,13 @@ def __init__(self, minimizer, result, p_names=None, prob_func=None,
         # check that there are at least 2 true variables!
         # check that all stderrs are sensible (including not None or NaN)
         nvars = 0
-        for par in self.fit_params:
+        for par in self.params.values():
```

Comment:
I don't think this looping to find the number of varying parameters is needed here, can this not just be ```nvars = len(self.result.var_names)``` ?

Reply:
No, we need the free params. If given a result class, simply asking for nfree should be fine, but when working the Minimizer directly, I think the code above is necessary.

Reply:
sorry, I am not sure I understand this. What do you mean by ```free parameters```, isn't that the same as ```varying parameters```? I think the exact loop you're doing here has already happened in ```minimizer.py-->prepare_fit```, see below for the relevant part: 
```
result.var_names = []

for name, par in self.result.params.items():
    if par.vary:
        result.var_names.append(name)
```

Reply:
After testing the proposed: No, we cannot do that. Since the user can change the vary between the fit and calling conf_interval.

Reply:
I am not sure I agree that a user can (or should) change the ```vary``` after the fit and before calculating the confidence intervals. As far as I can tell from the documentation and the code, the ```ConfidenceInterval``` class takes to initialization a ```Minimizer``` and  ```MinimizerResult``` class. How come we can not take the number of varying/free fitting parameters from the ```MinimizerResult```?

Reply:
okay, still think one should be using the ```MinimizerResult``` from a fit to calculate the ```ConfidenceInterval```. But, it doesn't take much time (or extra code) to just calculate the number of varying parameters again in cases someone doesn't do this; so sure, let's keep it as you suggested.